grails-app/services/org/pih/warehouse/product/ProductMergeService.groovy

```diff
@@ -140,6 +191,7 @@ class ProductMergeService {
             // Swap assemblyProduct to primary
             obsoletedComponent.assemblyProduct = primary
             // Note: needs flush because of "User.locationRoles not processed by flush"
+            // (this was an issue on grails 1.3.9, might not be a case anymore)
```

Comment:
nitpicky and not related/not important now, but I'm actually curious why `User.locationRoles` would even have an impact on those saves + if we were to remove the flush, we could get rid of those explicit saves at all, as we update an existing entity and all of the changes are tracked by Hibernate